File: legacy/legacy-apis/Psi.Models.ClientConfigurationModels/Admin/AdminSettings.cs
namespace Psi.Data.Models.ClientConfigurationModels.Admin
{
    public class AdminSettings : SettingsBaseHelper
    {
        private AdminAccountSettings _accountSettings;
        private Enrollment _enrollment;
        private MemberSettings _memberSettings;
        private DeviceSettings _deviceSettings;
        private MfaSettings _mfaSettings;
        private AuditLogs _auditLogs;
        private SmsSettings _smsSettings;
        private MemberView _memberView;
        private TransferSettings _transferSettings;

        public AdminSettings(ISettingsBase settingsBase) : base(settingsBase)
        {
        }

        [SettingKey("X.App.HomeBanking.AdminPasswordExpires")]
        public bool DoesAdminPasswordExpire
        {
            get => GetBoolValue();
            set => SetValue(value);
        }

        public AdminAccountSettings Account
        {
            get => _accountSettings ?? (_accountSettings = new AdminAccountSettings(SettingsBase));
            set => _accountSettings = value;
        }

        public Enrollment Enrollment
        {
            get => _enrollment ?? (_enrollment = new Enrollment(SettingsBase));
            set => _enrollment = value;
        }

        public MemberSettings Member
        {
            get => _memberSettings ?? (_memberSettings = new MemberSettings(SettingsBase));
            set => _memberSettings = value;
        }

        public DeviceSettings Device
        {
            get => _deviceSettings ?? (_deviceSettings = new DeviceSettings(SettingsBase));
            set => _deviceSettings = value;
        }

        public MfaSettings Mfa
        {
            get => _mfaSettings ?? (_mfaSettings = new MfaSettings(SettingsBase));
            set => _mfaSettings = value;
        }

        public AuditLogs AuditLogs
        {
            get => _auditLogs ?? (_auditLogs = new AuditLogs(SettingsBase));
            set => _auditLogs = value;
        }

        public SmsSettings SmsSettings
        {
            get => _smsSettings ?? (_smsSettings = new SmsSettings(SettingsBase));
            set => _smsSettings = value;
        }

        public MemberView MemberView
        {
            get => _memberView ?? (_memberView = new MemberView(SettingsBase));
            set => _memberView = value;
        }

        public TransferSettings Transfer
        {
            get => _transferSettings ?? (_transferSettings = new TransferSettings(SettingsBase));
            set => _transferSettings = value;
        }
    }
}

File: workspace/legacy-analyzer/output/infrastructure/models/Admin/AdminSettings.ts
// Generated imports
import { AdminAccountSettings } from './AdminAccountSettings';
import { Enrollment } from '../Enrollment';
import { MemberSettings } from '../MemberSettings';
import { DeviceSettings } from '../DeviceSettings';
import { MfaSettings } from '../MfaSettings';
import { AuditLogs } from '../AuditLogs';
import { SmsSettings } from '../SmsSettings';
import { MemberView } from '../MemberView';
import { TransferSettings } from '../TransferSettings';

export interface AdminSettings {
    /** @settingKey X.App.HomeBanking.AdminPasswordExpires */
    doesAdminPasswordExpire: boolean;
    adminAccountSettings: AdminAccountSettings;
    enrollment: Enrollment;
    memberSettings: MemberSettings;
    deviceSettings: DeviceSettings;
    mfaSettings: MfaSettings;
    auditLogs: AuditLogs;
    smsSettings: SmsSettings;
    memberView: MemberView;
    transferSettings: TransferSettings;
}
